oc_cdtapi/RundeckAPI.py

```diff
@@ -0,0 +1,599 @@
+
+    def scm__enable(self, project, integration, plugin_type, enable=False):
+        """
+        Turn on/off SCM for a project
+        :param str project: project name
+        :param str integration: one of: "import", "export"
+        :param str plugin_type: plugin to configure
+        :param bool enable: disable if set to False, enable otherwise
+        :return dict:
+        """
+        self._logger.info(
+                " ".join([
+                    'Enable' if enable else 'Disable',
+                    f"SCM for project [{project}], integration [{integration}], plugin_type [{plugin_type}]"]))
+
+        # checking arguments
+        if not project:
+            raise ValueError("Project name is mandatory")
+
+        if integration not in ["import", "export"]:
+            raise ValueError(f"Unsupported SCM integration type: [{integration}]")
+
+        if not plugin_type:
+            raise ValueError("Plugin type is mandatory")
+        
+        _req = ["project", project, "scm", integration, "plugin", plugin_type, "enable" if enable else "disable"]
+
+        return self.post(_req, headers=self.headers, cookies=self.cookies).json()
+
+    def scm__get_action_inputs(self, project, integration, action):
+        """
+        Get items needed to be performed actions on
+        :param str project: project name
+        :param str integration: one of: "import", "export"
+        :param str action: action to perform
+        :return dict:
+        """
+        self._logger.info(
+                f"Get SCM actions input list for project [{project}], integration [{integration}], action [{action}]")
+
+        # checking arguments
+        if not project:
+            raise ValueError("Project name is mandatory")
+
+        if integration not in ["import", "export"]:
+            raise ValueError(f"Unsupported SCM integration type: [{integration}]")
+
+        if not action:
+            raise ValueError("Action is mandatory")
+        
+        _req = ["project", project, "scm", integration, "action", action, "input"]
+
+        return self.get(_req, headers=self.headers, cookies=self.cookies).json()
+
+    def scm__action_perform(self, project, integration, action, action_data):
+        """
+        SCM import/export all project items
+        :param str project: project name
+        :param str integration: one of: "import", "export"
+        :param str action: action to perform
+        :param dict action_data: what to do
+        :return dict:
+        """
+        self._logger.info(" ".join([f"Perform SCM actions for project [{project}], integration [{integration}], action [{action}].",
+                                    f"Action data: {action_data}"]))
+
+        if not action_data:
+            self._logger.info(f"Emtpy action data, nothing to do")
+            return dict()
+
+        # checking arguments
+        if not project:
```

Comment:
This check is performed a number of times. I'd suggest some "check params" method, see comment about integration as well.